app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/activities/SearchActivity.java

```diff
@@ -0,0 +1,179 @@
+            @Override
+            public void onLoadMore(int page, int totalItemsCount) {
+                if (page < MAX_PAGE) {
+                    onImageSearch(page - 1);
+                }
+            }
+        });
+
+        imageResults = new ArrayList<>();
+        imageAdapter = new ImageResultArrayAdapter(this, imageResults);
+        gvResults.setAdapter(imageAdapter);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_search, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.miAdvancedSearch) {
+            FragmentManager fm = getSupportFragmentManager();
+            ImageFilterDialog imageFilterDialog= ImageFilterDialog.newInstance("Prueba", imageFilter);
+            imageFilterDialog.show(fm, "fragment_image_filter");
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    public void onImageSearch(int start) {
+
+        if (isNetworkAvailable()) {
+            client = new SearchClient();
+            query = etQuery.getText().toString();
+            startPage = start;
+            if (startPage == 0)
+                imageAdapter.clear();
+
+            if (!query.equals(""))
+                client.getSearch(query, startPage, imageFilter, this, new JsonHttpResponseHandler() {
+                            @Override
+                            public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
+                                try {
+                                    JSONArray imageJsonResults;
+                                    if (response != null) {
+                                        imageJsonResults = response.getJSONObject("responseData").getJSONArray("results");
+                                        imageAdapter.addAll(ImageResult.fromJSONArray(imageJsonResults));
+                                    }
+                                } catch (JSONException e) {
+                                    Toast.makeText(getApplicationContext(), "Invalid data received", Toast.LENGTH_SHORT).show();
+                                    e.printStackTrace();
+                                }
+                            }
+
+                            @Override
+                            public void onFailure(int statusCode, Header[] headers, String responseString, Throwable throwable) {
+                                super.onFailure(statusCode, headers, responseString, throwable);
```

Comment:
@edwinperaza you should handle the `onFailure` of the API call and give the user some visual queue of what happened